### ISSUE TITLE
bfloat16 not supported on Google Colab T4

### ISSUE THREAD

**Issue by DePasqualeOrg** [metavoiceio/metavoice-src#27]
I'm trying to get this working in Google Colab based on the info in the readme, but I get the following error:

```
  File "/content/metavoice-src/fam/llm/sample.py", line 696, in <module>
    smodel, llm_first_stage, llm_second_stage = build_models(
  File "/content/metavoice-src/fam/llm/sample.py", line 571, in build_models
    llm_first_stage = Model(
  File "/content/metavoice-src/fam/llm/sample.py", line 81, in __init__
    nullcontext() if device_type == "cpu" else torch.amp.autocast(device_type=device_type, dtype=ptdtype)
  File "/usr/local/lib/python3.10/dist-packages/torch/amp/autocast_mode.py", line 305, in __init__
    raise RuntimeError(
RuntimeError: Current CUDA Device does not support bfloat16. Please switch dtype to float16.
```

**Comment by sidroopdaska** [comment]
Yes, this is expected. bfloat16 is not supported on Nvidia T4s. See [here](https://github.com/vllm-project/vllm/issues/1400) for similar issue.

Would love if you can open a PR to handle this edge case?

PyTorch has an API to check for this: `torch.cuda.is_bf16_supported()`

**Comment by DePasqualeOrg** [comment]
I got the reference implementation script to run by including the flag `--dtype=float16`, but I got an out of memory error on the T4 GPU. I'm just trying to get this working on Colab, and I'm not sure how to proceed from here.

**Comment by sidroopdaska** [comment]
The model needs ~20GB of memory today. Therefore, a GPU with >= 24GB of memory will be required. Try using an a10 GPU
I'll add this to the README.md shortly.

**Comment by athenasaurav** [comment]
Hello @DePasqualeOrg 

I was able to run it in colab using A100 High Ram. 

Here is a link to [colab](https://colab.research.google.com/drive/1wydi9YWQe48HFPoVtP1ugfoxgLep_lIZ?usp=sharing)

**Comment by sidroopdaska** [comment]
Could you make a PR on setting up google colab up? It would be incredibly helpful to others

**Comment by athenasaurav** [comment]
Done @sidroopdaska
